database_sqlite.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"database/sql"
+	"log"
+	"os"
+	"time"
+)
+
+type SqliteDatabase struct {
+	*sql.DB
+}
+
+func sqliteInit(dbParams map[string]interface{}) (*SqliteDatabase, error) {
+
+	dbFilename, ok := dbParams["file"].(string)
+	if !ok {
+		dbFilename = "sqlite3.db"
+	}
+
+	db, err := sql.Open("sqlite3", dbFilename)
+	if err != nil {
+		return nil, err
+	}
+
+	statement := `
+		CREATE TABLE IF NOT EXISTS comments (
+			url varchar(2083) not null,
+			name varchar(200) not null,
+			comment varchar(3000) not null,
+			depth int not null,
+			time timestamp not null,
+			parent int
+		);
+	`
+	if _, err = db.Exec(statement); err != nil {
+		return nil, err
+	} else {
+
+		if os.Getenv("DEMO") == "true" {
```

Comment:
As a matter of fact, you can go ahead and remove the demo thing altogether; I'll use a different modified script to do this. There's no need for this to be in production code.